cmd/ctl/pkg/renew/renew.go

```diff
@@ -0,0 +1,212 @@
+	cmclient "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+)
+
+var (
+	long = templates.LongDesc(i18n.T(`
+	Mark cert-manager Certificate resources for manual renewal.
+`))
+
+	example = templates.Examples(i18n.T(`
+		# Renew the Certificates named 'my-app' and 'vault' in the current context namespace.
+		ctl renew my-app vault
+
+		# Renew all Certificates in the 'kube-system' namespace.
+		ctl renew --namespace kube-system --all
+
+		# Renew all Certificates in all namespaces that have the label 'app=my-service'.
+		ctl renew --all-namespaces -l app=my-service`))
+)
+
+// Options is a struct to support renew command
+type Options struct {
+	// The Namespace that the Certificate to be renewed resided in
+	Namespace  string
+	CMClient   cmclient.Interface
+	RestConfig *restclient.Config
+
+	LabelSelector string
+	All           bool
+	AllNamespaces bool
+
+	genericclioptions.IOStreams
+}
+
+// NewOptions returns initialized Options
+func NewOptions(ioStreams genericclioptions.IOStreams) *Options {
+	return &Options{
+		IOStreams: ioStreams,
+	}
+}
+
+// NewCmdRenew returns a cobra command for renewing Certificates
+func NewCmdRenew(ioStreams genericclioptions.IOStreams, factory cmdutil.Factory) *cobra.Command {
+	o := NewOptions(ioStreams)
+
+	cmd := &cobra.Command{
+		Use:     "renew",
+		Short:   "Mark a Certificate for manual renewal",
+		Long:    "Mark a Certificate for manual renewal",
+		Example: example,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(factory))
+			cmdutil.CheckErr(o.Validate(args))
+			cmdutil.CheckErr(o.Run(args))
+		},
+	}
+
+	cmd.Flags().StringVarP(&o.LabelSelector, "selector", "l", o.LabelSelector, "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
+	cmd.Flags().BoolVarP(&o.AllNamespaces, "all-namespaces", "A", o.AllNamespaces, "If present, mark Certificates across namespaces for manual renewal. Namespace in current context is ignored even if specified with --namespace.")
+	cmd.Flags().BoolVar(&o.All, "all", o.All, "Renew all Certificates in the given Namespace, or all namespaces with --all-namespaces enabled.")
+
+	return cmd
+}
+
+// Validate validates the provided options
+func (o *Options) Validate(args []string) error {
+	if len(o.LabelSelector) > 0 && len(args) > 0 {
+		return errors.New("cannot specify Certificate names in conjunction with label selectors")
+	}
+
+	if len(o.LabelSelector) > 0 && o.All {
+		return errors.New("cannot specify label selectors in conjunction with --all flag")
+	}
+
+	if o.All && len(args) > 0 {
+		return errors.New("cannot specify Certificate names in conjunction with --all flag")
+	}
+
+	return nil
+}
+
+// Complete takes the command arguments and factory and infers any remaining options.
+func (o *Options) Complete(f cmdutil.Factory) error {
+	var err error
+	o.Namespace, _, err = f.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
```

Comment:
Ah, does the `cmdutil.Factory` handle namespace flag registration? Perhaps add a comment on the field to make this clearer 😄